### PR TITLE
Task/07 api

### PR DESCRIPTION
**Pull request by VasylBiliak**
Acceptance Criteria for 1
 
- [x] Mentor is added to PR
- [x]  AC copied to PR description
- [x]  Link to deployed version is added to PR
- [x]  ProductPage fetches data from the API using either fetch or axios
- [x]  Pagination is implemented using query parameters
- [x]  Search functionality is implemented using query parameters, with the search triggered by a search button
- [x]  Category filtering is implemented using query parameters, resetting the page to the first page on change
- [x]  A loading spinner or message is displayed while awaiting API requests
- [x]  If the API response is an empty

 array or an error occurs, a "No products" message is displayed instead of a blank screen
Acceptance Criteria for 1* 

- [x] Pagination is hidden on mobile view
- [x]  Infinite scroll is implemented for mobile view, either manually or using a library
- [x]  Error notifications are implemented using axios interceptors, with component libraries for toastr notifications

deployed version
https://vasylbiliak.github.io/fe-react-2024

### CODE REVIEW

File: src/components/SearchBar/SearchBar.component.tsx [hunk 1]
```diff
@@ -46,31 +53,17 @@ function SearchBar({ setFilter, setSort, setSearchQuery }: SearchBarProps) {
...
-                    Clothes
-                </button>
-            </div>
-            <div className={styles.sort_by}>
-                <span className={styles.sort_by__text}>Sort by:</span>
-                <CustomSelector setSort={setSort} />
+                {categories.map((category) => (
+                    <button
+                        key={category.id}
+                        className={`${styles.filter__btn} ${selectedFilter === category.id.toString() ? styles.filters_btn_active : ''}`}
```
Comment (vadym-zinchenko-moc) [inline src/components/SearchBar/SearchBar.component.tsx]: for cases like this better use `clsx`

File: src/components/SearchBar/SearchBar.component.tsx [hunk 2]
```diff
@@ -11,15 +12,21 @@ interface SearchBarProps {
...
-    const [selectedFilters, setSelectedFilters] = useState<string[]>([]);
+const categories: Readonly<Pick<Category, 'id' | 'name'>[]> = [
+    { id: 1, name: 'Clothes' },
+    { id: 2, name: 'Electronics' },
+    { id: 3, name: 'Furniture' },
+    { id: 4, name: 'Shoes' },
+    { id: 5, name: 'Miscellaneous' },
+];
+
+export function SearchBar({ setFilter, setSort, setSearchQuery }: SearchBarProps) {
```
Comment (vadym-zinchenko-moc) [inline src/components/SearchBar/SearchBar.component.tsx:25]: Screenshot 2024-07-06 at 18 49 57

need rewrite search bar according design for mobile version

File: src/components/SearchBar/SearchBar.component.tsx [hunk 3]
```diff
@@ -11,15 +12,21 @@ interface SearchBarProps {
     setSearchQuery: (query: string) => void;
 }
 
-function SearchBar({ setFilter, setSort, setSearchQuery }: SearchBarProps) {
-    const [selectedFilters, setSelectedFilters] = useState<string[]>([]);
+const categories: Readonly<Pick<Category, 'id' | 'name'>[]> = [
```
Comment (vadym-zinchenko-moc) [inline src/components/SearchBar/SearchBar.component.tsx:17]: Screenshot 2024-07-06 at 18 49 22

when I click two times on same filter I don't see products, seems like bug

File: src/components/SearchBar/SearchBar.component.tsx [hunk 4]
```diff
@@ -11,15 +12,21 @@ interface SearchBarProps {
...
-        setSelectedFilters((previousFilters) =>
-            previousFilters.includes(filter) ? previousFilters.filter((f) => f !== filter) : [...previousFilters, filter],
-        );
-        setFilter(filter);
+    const handleFilterChange = (filterID: string) => {
+        setSelectedFilter(filterID);
+        return setFilter(filterID);
     };
 
     const handleSearchChange = (event: React.ChangeEvent<HTMLInputElement>) => {
```
Comment (vadym-zinchenko-moc) [inline src/components/SearchBar/SearchBar.component.tsx:39]: Screenshot 2024-07-06 at 18 48 57

when I write in lowercase name of product I don't see it. Need to fix it. Need to compare names in lowercase

**Review by vadym-zinchenko-moc — CHANGES_REQUESTED** [review]

File: src/components/SearchBar/SearchBar.component.tsx (hunk 4 above)
Comment (VasylBiliak) [reply src/components/SearchBar/SearchBar.component.tsx:39]: I've investigated the issue where searching for product names in lowercase doesn't yield results. It turns out that the backend is case-sensitive when handling product name searches. To resolve this, the backend should ideally normalize the case of the search input and product names to ensure case-insensitive searching.
Screenshot_1

Comment (VasylBiliak) [reply src/components/SearchBar/SearchBar.component.tsx:39]: Screenshot_2

**Review by vadym-zinchenko-moc — APPROVED** [review]

**Pull request merged**